### DISCUSSION TITLE
How to manually abort a query / mutation using auto-generated hooks

### DISCUSSION THREAD

**Discussion by sebastienlabine** [reduxjs/redux-toolkit#1313] [Q&A] [answered]
Let's say I have a auto-generated hook for uploading a document.

```js
const [upload, { isLoading, isSuccess, isError, error }] = useUploadDocumentMutation();
 ```

How do I manually trigger an abort signal for this hook?

**Comment by sebastienlabine** [comment, marked as answer]
Found it.

```
upload().abort();
```

**Reply by phryneas** [reply]
Yup, but be aware that this behaviour will lead your query into a "error" state. Generally I'd probably not bother aborting - the request has already been sent to the server and unless you are long-term streaming a response, you probably won't have any real benefits from aborting.

**Reply by sebastienlabine** [reply]
is there a reason why we can't cancel the network call? I feel like it's should trigger the abort signal that can then be handled by a custom base query.

**Reply by phryneas** [reply]
It should. I nowhere said that this won't work.
